adam/perception/high_level_semantics_situation_to_developmental_primitive_perception.py

```diff
@@ -763,8 +771,10 @@ def _perceive_ground_relations(
         for situation_object in self._situation.all_objects:
             if situation_object.ontology_node != GROUND:
                 if self._objects_to_perceptions[situation_object] in objects_to_relations:
-                    # If this object is not on anything else, it should be on the ground
+                    # If this object is not on anything else, it should be on the ground, unless it's explicitly
+                    # specified to be unsupported
                     object_is_on_or_in_something = False
+                    supported = True
```

Comment:
As long as we are fixing this, we might as well fix #706 as well.  The strategy for doing this:

- first check for the negated "on the ground" relation and bail if it is present
- then initialized `supported=False` and loop over the relations. If it is asserted to be in contact with any other object, set `supported=True`
- make the object on the ground if `supported` is `False`

Reply:
don’t we already do 706 in lines 780-790?